### ISSUE TITLE
error traversing trivial doc

### ISSUE THREAD

**Issue by alexch-m** [jbeder/yaml-cpp#972]
```
const char* text =
   "network:\n" \
   "  version: 2\n" \
   "  renderer: NetworkManager\n";

YAML::Node doc = YAML::Load(text);
for (YAML::const_iterator it = doc.begin(); it != doc.end(); ++it)
{
   auto type = it->Type();
}
```
it->Type() raised exception because of invalid node.
"terminate called after throwing an instance of 'YAML::InvalidNode'
  what():  invalid node; this may result from using a map iterator as a sequence iterator, or vice-versa"

...is there any WORKABLE yaml cpp library?

**Comment by jbeder** [comment]
The root node in your example is a map and out, so you need to access it through the 'first' and 'second' fields of the iterator.￼

**Comment by alexch-m** [comment]
Thank you. It would be nice to have a tree traversal example, I mean an implementation of
`void PrintNodes(const YAML::Node& node, uint level)`
I've created such but only after your hint.
